openquake/commonlib/readinput.py

```diff
@@ -414,11 +412,9 @@ def get_source_models(oqparam, source_model_lt, sitecol=None):
                 trt_model.gsims = gsim_lt.values[trt_model.trt]
         else:
             gsim_lt = logictree.DummyGsimLogicTree()
-        weight = (rlz.weight if oqparam.calculation_mode == 'event_based'
-                  else rlz.weight / num_samples)
+        weight = rlz.weight / num_samples
```

Comment:
It cannot happen. num_samples is built by counting the source model realizations, so at minimum it is 1.